dynd/nd/functional.pyx

```diff
@@ -70,7 +70,7 @@ cdef public object _jit(object func, intptr_t nsrc, const _type *src_tp):
 
     # Compile the function with Numba
     func.compile(signature)
-    compile_res = func._compileinfos[signature]
+    compile_res = func.overloads[signature]
```

Comment:
This isn't enough on its own. With the latest numba there's another error at line 81 where there's no attribute `jit_codegen`.